contracts/src/v0.8/dev/ocr2dr/AuthorizedOriginReceiver.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "../interfaces/AuthorizedReceiverInterface.sol";
+
+/**
+ * @notice Modified AuthorizedReciever abstract for use on the OCR2DROracle contract to limit usage
+ * @notice Uses tx.origin instead of msg.sender because the client contract sends messages to the Oracle contract
+ * @dev NOTE: Use the following steps to use for deployments. Do not leave these changes in the repository code.
+ * @dev To use:
+ * 1. Make the Oracle contract ownable, to control who can set the authorized senders
+ * ```
+ * import "../../ConfirmedOwner.sol";
+ * ...
+ * contract OCR2DROracle is OCR2DROracleInterface, OCR2Base, ConfirmedOwner {
+ * ...
+ * constructor() OCR2Base(true) ConfirmedOwner(msg.sender) {}
+ * ```
+ *
+ * 2. Extend OCR2DROracle.sol with this contract
```

Comment:
I'm guessing you'll do that in a separate PR...? Why do we need a comment here then?

Reply:
If we were to do those steps now we would have to modify every tests to set up the authorized senders to the oracle.
Then once beta was over we would have to remove this and undo all of the tests.

If you think it's important to have it in the repo I can spend the time to do that.

Reply:
So is the plan to always add this locally before deployment..? That's just a recipe for mistakes. And it won't be obvious what's deployed to prod.
How about we add a constructor flag "authorizeSenders" that can be switched off in tests to disable the check?

Reply:
This sounds good to me. I'll make the changes today.